src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -114,6 +122,18 @@ public bool HasErrors()
             }
             return declarationsByAssignment.ToImmutableDictionary();
         }
+        /// <summary>
+        /// Gets all diagnostics raised by loading Bicep config for this template.
+        /// </summary>
+        private IEnumerable<IDiagnostic> GetConfigDiagnostics()
+        {
+            var diagnosticWriter = ToListDiagnosticWriter.Create();
+            foreach (var builderFunc in Configuration.DiagnosticBuilders)
+            {
+                // This diagnostic does not correspond to any specific location in the template, so just use the first character span.
+                yield return builderFunc(DiagnosticBuilder.ForPosition(new Parsing.TextSpan(0, 1)));
```

Comment:
It would probably be good to converge on an approach for text spans that don't fit a position in the document - in some places I see `TextSpan(0, 1)` - in others I see `TextSpan(0, 0)`. Should we create a static property on `TextSpan` - e.g. `TextSpan.DocumentStart`?

Reply:
I'm not sure if using a `TextSpan(0, 1)` for an empty file would cause any errors, so `TextSpan(0, 0)` is probably the safer choice.

Reply:
Good point!